dev_guide/app_tutorials/ruby_on_rails.adoc

```diff
@@ -0,0 +1,344 @@
+----
+
+In case you would like to also set password for the database administrator, append the previous command with:
+----
+-e POSTGRESQL_ADMIN_PASSWORD=admin_pw
+----
+
+To watch the progress of this command type:
+
+----
+$ oc get pods --watch
+----
+
+=== Creating frontend service
+
+[WARNING]
+====
+*ToBeDone*:
+We in OpenShift need a way how specify deployment hooks upon the deployment process. The use case in the application is the set the databse migration(and maybe also the DB seeding) upon the deployment. For now the hooks can be specified only in the template.
+====
+
+To bring your application to OpenShift, you need to specify a repository in which your application lives, using once again the link:../../dev_guide/new_app.html[oc new-app] command, in which you will need to specify database related link:../../db_images/postgresql.html#environment-variables[environment variables] we setup in the link:./ruby_on_rails.html#creating-database-service[creating database service] section:
+
+----
+$ oc new-app path/to/source/code --name=rails-app -e POSTGRESQL_USER=username -e POSTGRESQL_PASSWORD=password -e POSTGRESQL_DATABASE=db_name
+----
+
+With this command OpenShift will fetch the source code, setup the Builder image, link:../../dev_guide/builds.html[build] your application image and also will deploy the newly created image together with the specified link:../../db_images/postgresql.html#environment-variables[environment variables]. Application is going to be named `rails-app`.
+
+You can verify the environment variables have been added by viewing the JSON document of the `rails-app` link:../../dev_guide/deployments.html[DeploymentConfig]:
+
+----
+$ oc get dc rails-app -o json
+----
+
+You should see the following section:
+
+----
+env": [
+    {
+        "name": "POSTGRESQL_USER",
+        "value": "username"
+    },
+    {
+        "name": "POSTGRESQL_PASSWORD",
+        "value": "password"
+    },
+    {
+        "name": "POSTGRESQL_DATABASE",
+        "value": "db_name"
+    }
+],
+----
+
+To check the link:../../dev_guide/builds.html[build] process use the link:../../dev_guide/builds.html#accessing-build-logs[build-logs] command:
+
+----
+$ oc build-logs rails-app-1
+----
+
+Once the link:../../dev_guide/builds.html[build] is complete, you can look at the running link:../../architecture/core_concepts/pods_and_services.html#pods[pods] in OpenShift
+
+----
+$ oc get pods
+----
+
+You should see a line starting with myapp-(#number)-(some hash) and that is your application running in OpenShift.
+
+=== Creating a route for your application
+
+To expose a service by giving it an externally-reachable hostname like `www.example.com` use OpenShift link:../../dev_guide/routes.html[route]. In your case you need to expose the frontend service by typing:
+
+----
+$ oc expose service rails-app
+----
+
+[WARNING]
+====
+TBD: We in OpenShift need to document how to setup DNS on host when using OpenShift in VM to be able to  access the application.
```

Comment:
I'd just put some general text in about it being the user's responsibility to ensure the hostname they specified resolves to the ip address of their router, and link to some appropriate openshift documentation about the router and DNS.